src/WebJobs.Script/Binding/BuiltinExtensionBindingProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Extensibility;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    /// <summary>
+    /// We have a backwards compat requirement to whitelist #r references to certain "builtin" dlls.
+    /// Hook into #r resolution pipeline and apply the whitelist.
+    /// </summary>
+    internal class BuiltinExtensionBindingProvider : ScriptBindingProvider
```

Comment:
I moved a bunch of this stuff out of ScriptHost into separate Types to improve cohesiveness and code readability